src/main/java/org/jabref/gui/exporter/ExportCustomizationDialogViewModel.java

```diff
@@ -0,0 +1,71 @@
+import javafx.collections.FXCollections;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.logic.exporter.TemplateExporter;
+import org.jabref.preferences.CustomExportList;
+import org.jabref.preferences.JabRefPreferences;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExportCustomizationDialogViewModel extends BaseDialog<Void> {
+
+    //The class vars might need to be reordered
+
+    private final SimpleListProperty<ExporterViewModel> exporters = new SimpleListProperty<>(FXCollections.observableArrayList());
+    private final int size; //final?  Or you don't need this and just use a a while loop
+
+    //Indices within which export format information is stored within JabRefPreferences
+    private static final int EXPORTER_NAME_INDEX = 0;
+    private static final int EXPORTER_FILENAME_INDEX = 1;
+    private static final int EXPORTER_EXTENSION_INDEX = 2;
+
+    private final PreferencesService preferences;
+    private final DialogService dialogService;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomExportList.class);
+
+    //Other variable declarations here
+
+    //Also write tests for all of this
+
+    public ExportCustomizationDialogViewModel(DialogService dialogService) {
+        this.dialogService = dialogService;
+
+        int i = 0;
+        //raname var "s"?
+        List<String> s;
+        while (!((s = preferences.getStringList(JabRefPreferences.CUSTOM_EXPORT_FORMAT + i)).isEmpty())) {
+            //Shuold createFormat be in logic?  Check how it works in CustomExportList
+            Optional<ExporterViewModel> format = createFormat(s.get(EXPORTER_NAME_INDEX), s.get(EXPORTER_FILENAME_INDEX), s.get(EXPORTER_EXTENSION_INDEX), layoutPreferences, savePreferences);
+            if (format.isPresent()) {
+                exporters.add(format.get()); //put was changed to add becuase we are not dealing with Map as in CustomExpoertList
+            } else {
+                String customExportFormat = preferences.get(JabRefPreferences.CUSTOM_EXPORT_FORMAT + i);
+                LOGGER.error("Error initializing custom export format from string " + customExportFormat);
+            }
+            i++;
+        }
+        //ExporterViewModel will be organized as a singular version of what now is CustomExportDialog, which
+        //currently stores all the exporters in a class var.  Each ViewModel will have one exporter and associated data, and
+        //the class var exporters will be a list of them
+        //You will have to write properites into ExpoerterViewModel that get all the relevant information about the exporter
+        //in addition to a property for the exporter itself
+
+    }
+
+    //possibly not necessary, and if so getSortedList will have to return the correct type, not Eventlist<List<String>>
+    public List<ExporterViewModel> loadExporters() {
```

Comment:
You might don't need the gui-wrapper `ExporterViewModel` around `Exporter`. In general such a wrapper is needed to adapt the properties of the model to display it to the user. But, here, the data of an exporter is relatively simple and the already in a suitable format for the gui.

Reply:
I have put the data in the GUI wrapper `ExporterViewModel` for now.  How would I create the `TableView` and `TableColumns` without a `ViewModel` for the data?  All the examples I could find in JabRef and online had some sort of `ViewModel`.

Reply:
You can also simply use `Exporter` as the datatype for the TableView. But if you have already created a ViewModel for it, keep it. It is probably the cleaner approach and more versatile for the future.

Reply:
Okay.  I've finished, but not tested, most of the `ViewModel`.  I just have to add the dialog to add or modify the exporter, and then create the View and FXML.  Unless there's something I forgot...